core/xchaincore.go

```diff
@@ -776,7 +780,18 @@ func (xc *XChainCore) PostTx(in *pb.TxStatus, hd *global.XContext) (*pb.CommonRe
 		xc.log.Debug("refused a connection at function call GenerateTx", "logid", in.Header.Logid)
 		return out, false
 	}
-
+	txid := in.GetTxid()
+	if txid == nil {
+		out.Header.Error = pb.XChainErrorEnum_CONNECT_REFUSE // 拒绝
+		xc.log.Debug("refused a tx with the txid being nil", "logid", in.GetHeader().GetLogid())
+		return out, false
+	}
+	if _, exist := xc.txidCache.Get(string(txid)); exist {
```

Comment:
The conversion from []byte to string has a copy overhead, so try to avoid multiple copies. 
`string(txid)` here.